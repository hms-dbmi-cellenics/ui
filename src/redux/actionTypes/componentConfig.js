/**
 * Sets plot configs and data.
 */
const COMPONENT_CONFIG = 'componentConfig';
const LOAD_CONFIG = `${COMPONENT_CONFIG}/load`;
const UPDATE_CONFIG = `${COMPONENT_CONFIG}/update`;
const RESET_CONFIG = `${COMPONENT_CONFIG}/reset`;
const SAVE_CONFIG = `${COMPONENT_CONFIG}/save`;
const PLOT_DATA_LOADED = `${COMPONENT_CONFIG}/plotDataLoaded`;
const PLOT_DATA_LOADING = `${COMPONENT_CONFIG}/plotDataLoading`;
const PLOT_DATA_ERROR = `${COMPONENT_CONFIG}/plotDataError`;

export {
  LOAD_CONFIG, UPDATE_CONFIG, RESET_CONFIG, SAVE_CONFIG,
  PLOT_DATA_LOADED, PLOT_DATA_LOADING, PLOT_DATA_ERROR,
};
