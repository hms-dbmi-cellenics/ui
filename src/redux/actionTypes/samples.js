const SAMPLES = 'samples';

/**
 * Save sample.
 */
const SAMPLES_CREATE = `${SAMPLES}/create`;

/**
 * Update sample.
 */
const SAMPLES_UPDATE = `${SAMPLES}/update`;

/**
 * Delete sample.
 */
const SAMPLES_DELETE_API_V2 = `${SAMPLES}/deleteApiV2`;

/**
 * Delete sample following apiv1 version, remove when doing the final step of cleaning up v1 code.
 */
const SAMPLES_DELETE_API_V1 = `${SAMPLES}/deleteApiV1`;

/**
 * Update files in sample.
 */
const SAMPLES_FILE_UPDATE = `${SAMPLES}/fileUpdate`;

/**
 * Load sample.
 */
const SAMPLES_LOADED = `${SAMPLES}/loaded`;

/**
 * Saving samples.
 */
const SAMPLES_SAVING = `${SAMPLES}/saving`;

/**
 * Samples saved successfully.
 */
const SAMPLES_SAVED = `${SAMPLES}/saved`;

/**
 * Error loading/saving sample.
 */
const SAMPLES_ERROR = `${SAMPLES}/error`;

/**
 * Delete metada from sample.
 */
const SAMPLES_METADATA_DELETE = `${SAMPLES}/metadataDelete`;

const SAMPLES_LOADING = `${SAMPLES}/loading`;
export {
  SAMPLES_CREATE,
  SAMPLES_UPDATE,
  SAMPLES_DELETE_API_V2,
  SAMPLES_DELETE_API_V1,
  SAMPLES_FILE_UPDATE,
  SAMPLES_LOADED,
  SAMPLES_SAVING,
  SAMPLES_ERROR,
  SAMPLES_SAVED,
  SAMPLES_METADATA_DELETE,
  SAMPLES_LOADING,
};
