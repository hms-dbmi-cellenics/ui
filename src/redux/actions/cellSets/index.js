import createCellSet from './createCellSet';
import deleteCellSet from './deleteCellSet';
import deleteCellClass from './deleteCellClass';
import runCellSetsClustering from './runCellSetsClustering';
import runCellSetsAnnotation from './runCellSetsAnnotation';
import runCellCycleScoring from './runCellCycleScoring';
import updateCellSetsClustering from './updateCellSetsClustering';

import loadCellSets from './loadCellSets';
import reorderCellSet from './reorderCellSet';
import updateCellSetProperty from './updateCellSetProperty';
import updateCellSetSelected from './updateCellSetSelected';
import setCellSetHiddenStatus from './setCellSetHiddenStatus';
import unhideAllCellSets from './unhideAllCellSets';

export {
  createCellSet,
  deleteCellSet,
  deleteCellClass,
  runCellSetsClustering,
  runCellSetsAnnotation,
  updateCellSetsClustering,
  loadCellSets,
  reorderCellSet,
  updateCellSetSelected,
  updateCellSetProperty,
  setCellSetHiddenStatus,
  unhideAllCellSets,
  runCellCycleScoring,
};
