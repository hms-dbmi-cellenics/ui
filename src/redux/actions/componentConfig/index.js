import loadPlotConfig from 'redux/actions/componentConfig/loadPlotConfig';
import loadComponentConfig from 'redux/actions/componentConfig/loadComponentConfig';
import updatePlotConfig from 'redux/actions/componentConfig/updatePlotConfig';
import savePlotConfig from 'redux/actions/componentConfig/savePlotConfig';
import fetchPlotDataWork from 'redux/actions/componentConfig/fetchPlotDataWork';
import updatePlotData from 'redux/actions/componentConfig/updatePlotData';
import resetPlotConfig from 'redux/actions/componentConfig/resetPlotConfig';

export {
  loadPlotConfig,
  loadComponentConfig,
  updatePlotConfig,
  savePlotConfig,
  fetchPlotDataWork,
  updatePlotData,
  resetPlotConfig,
};
