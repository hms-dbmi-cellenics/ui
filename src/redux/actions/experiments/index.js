import createExperiment from './createExperiment';
import saveExperiment from './saveExperiment';
import updateExperiment from './updateExperiment';
import loadExperiments from './loadExperiments';
import switchExperiment from './switchExperiment';
import reorderSamples from './reorderSamples';

export {
  createExperiment,
  saveExperiment,
  updateExperiment,
  loadExperiments,
  switchExperiment,
  reorderSamples,
};
