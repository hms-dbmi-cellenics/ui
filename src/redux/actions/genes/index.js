import loadGeneExpression from './loadGeneExpression';
import changeGeneSelection from './changeGeneSelection';
import loadPaginatedGeneProperties from './loadPaginatedGeneProperties';
import loadMarkerGenes from './loadMarkerGenes';
import loadDownsampledGeneExpression from './loadDownsampledGeneExpression';

export {
  loadGeneExpression,
  changeGeneSelection,
  loadPaginatedGeneProperties,
  loadMarkerGenes,
  loadDownsampledGeneExpression,
};
