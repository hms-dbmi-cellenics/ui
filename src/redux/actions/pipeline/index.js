import runQC from './runQC';
import runGem2s from './runGem2s';
import runSubsetExperiment from './runSubsetExperiment';
import runObj2s from './runObj2s';
import retry from './retry';

export {
  runQC,
  runGem2s,
  runSubsetExperiment,
  runObj2s,
  retry,
};
