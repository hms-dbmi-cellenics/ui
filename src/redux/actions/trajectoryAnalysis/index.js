import loadTrajectoryAnalysis from './loadTrajectoryAnalysis';

export {
  // eslint-disable-next-line import/prefer-default-export
  loadTrajectoryAnalysis,
};
