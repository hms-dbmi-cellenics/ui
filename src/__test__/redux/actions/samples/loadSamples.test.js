import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';

import initialSampleState from '../../../../redux/reducers/samples/initialState';
import { SAMPLES_ERROR, SAMPLES_LOADED } from '../../../../redux/actionTypes/samples';
import { loadSamples } from '../../../../redux/actions/samples';

enableFetchMocks();

const mockStore = configureStore([thunk]);

describe('loadSample action', () => {
  const experimentId = '1234';

  const initialState = {
    samples: {
      ...initialSampleState,
    },
  };

  const response = new Response(
    JSON.stringify(
      [
        {
          samples: {
            ids: ['sample-1', 'sample-2'],
            'sample-1': { name: 'sample-1' },
            'sample-2': { name: 'sample-2' },
          },
        }],
    ),
  );

  fetchMock.resetMocks();
  fetchMock.doMock();
  fetchMock.mockResolvedValue(response);

  it('Dispatches event correctly', async () => {
    const store = mockStore(initialState);
    await store.dispatch(loadSamples(experimentId));

    // LOAD SAMPLE
    const action1 = store.getActions()[1];
    expect(action1.type).toEqual(SAMPLES_LOADED);
  });

  it('Dispatches error correctly', async () => {
    fetchMock.mockReject(new Error('Failed fetching samples'));

    const store = mockStore(initialState);
    await store.dispatch(loadSamples(experimentId));

    // LOAD SAMPLE
    const action1 = store.getActions()[1];
    expect(action1.type).toEqual(SAMPLES_ERROR);
  });
});
