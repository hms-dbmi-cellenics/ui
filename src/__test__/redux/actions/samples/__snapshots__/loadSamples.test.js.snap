// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadSample action Dispatches error correctly 1`] = `
[
  undefined,
  {
    "error": "Error: An api error",
  },
]
`;

exports[`loadSample action Works correctly 1`] = `
{
  "samples": {
    "e03ef6ea-5014-4e57-aecd-59964ac9172c": {
      "createdDate": "2021-12-07 17:36:27.773+00",
      "experimentId": "1234",
      "files": {
        "barcodes10x": {
          "size": undefined,
          "upload": {
            "status": "uploaded",
          },
        },
        "features10x": {
          "size": undefined,
          "upload": {
            "status": "uploaded",
          },
        },
        "matrix10x": {
          "size": undefined,
          "upload": {
            "status": "uploaded",
          },
        },
      },
      "lastModified": "2021-12-07 17:38:42.036+00",
      "metadata": {
        "age": "BL",
        "timePoint": "BL",
      },
      "name": "BLp7",
      "options": {},
      "type": "10x",
      "uuid": "e03ef6ea-5014-4e57-aecd-59964ac9172c",
    },
  },
}
`;
